platforms/core-runtime/base-services/src/main/java/org/gradle/internal/classloader/MultiParentClassLoader.java

```diff
@@ -55,6 +56,11 @@ public MultiParentClassLoader(Collection<? extends ClassLoader> parents) {
         this.parents = new CopyOnWriteArrayList<ClassLoader>(parents);
     }
 
+    @Override
+    public String toString() {
+        return MultiParentClassLoader.class.getSimpleName() + "(" + StringUtils.join(parents, ", ") + ")";
```

Comment:
>I presume this change is needed for real-world errors to be readable, but this is not covered by tests.

That's a good catch. I've added a simple, though trivial, test for that to make sure I didn't mess up with the string formatting.

Reply:
Sorry I wasn't clear. I meant an actual integration test that would use whatever classloaders are setup by Gradle. Something the user might stumble upon, when working with Gradle. Its those "toString"s that are important to be meaningful.

The last time I saw such classloader confusion was in relation to init scripts that tried filtering Kotiln tasks by type, but it didn't work, because init script's runtime classpath differs from the one for project that actually applies the Kotlin plugin. Though, I am sure it's possible to find a simpler example. 

I think settings script classloader is not a parent of a project classloader, so we could use this to create an example of a problem.